File: backend authentication/src/config/testConnection.ts
import { supabase } from "./supabase";

async function testConnection() {
  console.log("Testing Supabase connection...");

  const { data, error } = await supabase
    .from("user_auth")
    .select("id")
    .limit(1);

  if (error) {
    console.error("Connection failed:", error.message);
    process.exit(1);
  }

  console.log("Supabase connected successfully!");
  console.log("user_auth table is accessible. Rows returned:", data?.length ?? 0);
  process.exit(0);
}

testConnection();

File: backend authentication/config/test_connection.py
from config.supabase_client import supabase


def test_connection():
    print("Testing Supabase connection...")

    response = supabase.from_("user_auth").select("id").limit(1).execute()

    if hasattr(response, "error") and response.error:
        print("Connection failed:", response.error.message)
        raise SystemExit(1)

    print("Supabase connected successfully!")
    print("user_auth table is accessible. Rows returned:", len(response.data or []))


if __name__ == "__main__":
    test_connection()
